sqlglot/optimizer/annotate_types.py

```diff
@@ -568,48 +568,85 @@ def _annotate_literal(self, expression: exp.Literal) -> exp.Literal:
 
         return expression
 
+    def _maybe_wrap_with_array(self, expression: E, array: bool = False) -> E:
+        if array:
```

Comment:
I suggest changing this to `_array_of` or something and then inlining the `array` check in the two places it's used right now.